### PR TITLE
[#223] Add CoAP X.509 support

### LINKED ISSUE #223: Enable coap x509 support for hono
In order to enable coap to use x509, the `hono-adapter-coap-vertx-secret.yaml` must add

```sh
application.yml: |
    hono:
      coap:
        keyPath: /etc/hono/key.pem
        certPath: /etc/hono/cert.pem
```

### LINKED ISSUE #224: [Hono] Readme documentation for extraSecretMounts misleading
Hi all,

first of all I am new to the kubernetes and helm world, therefore I hope the assumptions I make later on are correct. If not I appreciate a correction!

I am trying to use hono with enMasse and not the provided AMQP network. The readme of the hono chart has a section about that ([Integrating with an existing AMQP Messaging Network](https://github.com/eclipse/packages/tree/master/charts/hono#integrating-with-an-existing-amqp-messaging-network)) where it states the following:
```yaml
# do not deploy example AMQP Messaging Network
amqpMessagingNetworkExample:
  enabled: false

adapters:

  # mount (existing) Kubernetes secret which contains
  # credentials for connecting to AMQP network
  extraSecretMounts:
  - amqpNetwork:
      secretName: "my-secret"
      mountPath: "/etc/custom"

  # provide connection params
  # assuming that "my-secret" contains an "amqp-credentials.properties" file
  amqpMessagingNetworkSpec:
    host: my-custom.amqp-network.org
    port: 5672
    credentialsPath: "/etc/custom/amqp-credentials.properties"
  commandAndControlSpec:
    host: my-custom.amqp-network.org
    port: 5672
    credentialsPath: "/etc/custom/amqp-credentials.properties"
```

Using those instructions I generated my own values file looking like that:
```yaml
amqpMessagingNetworkExample:
  enabled: false

adapters:
  componentConfig:
    extraSecretMounts:
    - amqpNetwork:
        secretName: "mysecret"
        mountPath: "/etc/custom"

  amqpMessagingNetworkSpec:
    host: messaging-5355a0a.enmasse-infra
    port: 5672
    credentialsPath: /etc/custom/amqp-credentials.properties

  commandAndControlSpec:
    host: messaging-5355a0a.enmasse-infra
    port: 5672
    credentialsPath: /etc/custom/amqp-credentials.properties

  amqp:
    enabled: false
```
After installing the chart the adapters for http and mqtt produced with the following warning message:
`08:32:06.926 [vert.x-eventloop-thread-0] WARN  o.e.h.config.ClientConfigProperties - could not load client credentials for [messaging-5355a0a.enmasse-infra:5672, role: AMQP Messaging Network] from file [/etc/custom/amqp-credentials.properties]
java.io.FileNotFoundException: /etc/custom/amqp-credentials.properties (No such file or directory)`

After checking the /etc path in the pod the secrets seemed not to be mounted (the secret exists in my kubernetes cluster). 

After spending some time in the [_helpers.tpl](https://github.com/eclipse/packages/blob/master/charts/hono/templates/_helpers.tpl) file and trying to debug in there I found two inconsistencies with the redame:

1. The scope gets changed in the heper function [hono.container.secretVolumeMounts](https://github.com/eclipse/packages/blob/master/charts/hono/templates/_helpers.tpl#L486), which means the `extraSecretMounts` need to be inside an appending http or mqtt depending on the adapter.
2. The helper function can not get a valid name due to a misformed map[]. This can be changed erasing the '-' in front of the 'amqpNetwork' in my value.yaml file.

Therefore the resulting value file looks like this:
```yaml
amqpMessagingNetworkExample:
  enabled: false

adapters:
  http:
    extraSecretMounts:
      amqp-network:
        secretName: "mysecret"
        mountPath: "/etc/custom"
  mqtt:
    extraSecretMounts:
      amqp-network:
        secretName: "mysecret"
        mountPath: "/etc/custom"

  amqpMessagingNetworkSpec:
    host: messaging-5355a0a.enmasse-infra
    port: 5672
    credentialsPath: /etc/custom/amqp-credentials.properties

  commandAndControlSpec:
    host: messaging-5355a0a.enmasse-infra
    port: 5672
    credentialsPath: /etc/custom/amqp-credentials.properties

  amqp:
    enabled: false
```

Is that the correct way and needs the readme to be updated or am I doing something wrong?

### PR DESCRIPTION
**Pull request by calohmn**
This fixes #223.

### CODE REVIEW

**Comment by calohmn** [comment]
Postponed until Hono 1.8 gets used in the chart.

**Comment by sophokles73** [comment]
@calohmn I think that this can now be merged. Can you rebase and bump the chart version?

File: charts/hono/Chart.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ description: |
   interacting with them in a uniform way regardless of the device communication protocol.
 version: 1.8.0
 # Version of Hono being deployed by the chart
-appVersion: 1.8.0
+appVersion: 1.8.1
```
Comment (sophokles73) [inline charts/hono/Chart.yaml:20]: I guess you wanted to bump the *version* property instead ...

Comment (calohmn) [reply charts/hono/Chart.yaml]: yes, fixed now.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**